### PR TITLE
Adds /resolvedEnv endpoint

### PR DESCRIPTION
**Pull request by ttomsu**
A super handy debugging tool, `/resolveEnv` is different from the `/env` endpoint, which potentially lists the same value multiple times across different PropertySources. This endpoint prints the value once fully resolved.

Example output snippet from Gate:

```
{
  ...
  "server.address": "localhost",
  "server.port": "8084",
  "services.bakery.allowMissingPackageInstallation": "false",
  "services.bakery.baseUrl": "http://localhost:8087",
  "services.bakery.extractBuildDetails": "true",
  "services.bakery.host": "localhost",
  "services.bakery.port": "8087",
  "services.cassandra.cluster": "CASS_SPINNAKER",
  "services.cassandra.embedded": "false",
  "services.cassandra.host": "localhost",
  "services.cassandra.port": "9042",
  "services.clouddriver.aws.udf.enabled": "true",
  "services.clouddriver.baseUrl": "http://localhost:7002",
  "services.clouddriver.host": "localhost",
  "services.clouddriver.port": "7002",
  "services.deck.auth.enabled": "true",
  "services.deck.bakeryUrl": "http://localhost:8087",
  "services.deck.baseUrl": "https://localhost:9000",
  "services.deck.gateUrl": "https://localhost:8084",
  "services.deck.host": "localhost",
  "services.deck.port": "9000",
  "services.deck.timezone": "America/Los_Angeles",
  "services.default.host": "localhost",
  "services.default.primaryAccountName": "my-google-account",
  "services.default.protocol": "http",
  "services.docker.baseUrl": "http://my-docker-instance:7104",
  "services.docker.enabled": "true",
  "services.docker.primaryAccount.name": "my-google-account",
  "services.docker.primaryAccount.registry": "http://localhost:5000",
  "services.docker.primaryAccount.url": "http://my-docker-instance:7104",
  "services.docker.targetRepository": "my-docker-instance:5000",
  "services.dockerRegistry.baseUrl": "http://localhost:5000",
  "services.echo.baseUrl": "http://localhost:8089",
  "services.echo.cassandra.enabled": "true",
  "services.echo.cron.enabled": "true",
  "services.echo.cron.timezone": "America/Los_Angeles",
  "services.echo.enabled": "true",
  "services.echo.host": "localhost",
  "services.echo.inMemory.enabled": "false",
  "services.echo.notifications.hipchat.botName": "",
  "services.echo.notifications.hipchat.enabled": "false",
  "services.echo.notifications.hipchat.token": "",
  "services.echo.notifications.hipchat.url": "",
  "services.echo.notifications.mail.enabled": "false",
  "services.echo.notifications.mail.fromAddress": "",
  "services.echo.notifications.mail.host": "",
  "services.echo.notifications.slack.botName": "",
  "services.echo.notifications.slack.enabled": "false",
  "services.echo.notifications.slack.token": "",
  "services.echo.notifications.sms.account": "",
  "services.echo.notifications.sms.enabled": "false",
  "services.echo.notifications.sms.from": "",
  "services.echo.notifications.sms.token": "",
  "services.echo.port": "8089",
  "services.fiat.autoConfig": "false",
  "services.fiat.baseUrl": "http://localhost:7003",
  "services.fiat.enabled": "false",
  "services.fiat.host": "localhost",
  "services.fiat.port": "7003",
  "services.flapjack.enabled": "false",
  "services.front50.baseUrl": "http://localhost:8080",
  "services.front50.bucket_location": "",
  "services.front50.bucket_root": "front50",
  "services.front50.cassandra.enabled": "true",
  "services.front50.gcs.enabled": "false",
  "services.front50.host": "localhost",
  "services.front50.port": "8080",
  "services.front50.redis.enabled": "false",
  "services.front50.s3.enabled": "false",
  "services.front50.storage_bucket": "",
  "services.gate.baseUrl": "http://localhost:8084",
  "services.gate.host": "localhost",
  "services.gate.port": "8084",
  "services.igor.baseUrl": "http://localhost:8088",
  "services.igor.enabled": "true",
  "services.igor.host": "localhost",
  "services.igor.port": "8088",
  "services.jenkins.defaultMaster.baseUrl": "http://jenkins-1:80/jenkins",
  "services.jenkins.defaultMaster.name": "Jenkins",
  ...
}
```

@jtk54 @duftler @lwander @ewiseblatt @cfieber @ajordens, and anyone else who helps Slack users debug their installations, PTAL

### CODE REVIEW

File: kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  public ResolvedEnvironmentEndpoint() {
+    super("resolvedEnv")
+  }
+
+  @Override
+  Map<String, Object> invoke() {
+    def result = [:]
+    Environment environment = getEnvironment()
+    getPropertyKeys().each {
```
Comment (duftler) [inline kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy]: Can use `.collectEntries` here.

File: kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      result.put(it, environment.getProperty(it))
+    }
+    return result
+  }
+
+  /**
+   * Impl paritially copied from
```
Comment (duftler) [inline kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy]: s/paritially/partially

**Comment by ewiseblatt** [comment]
Looks very useful.

Does this endpoint need to be suppressible?
What happens with sensitive keys (password values)?

**Comment by cfieber** [comment]
the built in env endpoint suppresses passwords, would be worth doing that here too?

`org.springframework.boot.actuate.endpoint.Sanitizer` is the implementation (package private but probably worth copying or seeing if there is a way to extend EnvironmentEndpoint?)

File: kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy (hunk 1 above)
Comment (ttomsu) [inline kork-web/src/main/groovy/com/netflix/spinnaker/config/ResolvedEnvironmentEndpoint.groovy]: Done.

**Comment by ttomsu** [comment]
Moved class to spring package to make `Sanitizer` accessible. Now all properties ending with "secret", "key", or "password" will appear as "*****". Sample output:

```
  "security.user.password": "******",
```

**Pull request merged**